modules/aggregation-layers/src/heatmap-layer/heatmap-layer-utils.js

```diff
@@ -0,0 +1,113 @@
+import {experimental, createIterable} from '@deck.gl/core';
+const {count} = experimental;
+
+// Parse input data to build positions, wights arrays.
+export function parseData(data, getPosition, getWeight) {
+  const pointCount = count(data);
+
+  const positions = new Float32Array(pointCount * 3);
+  const weights = new Float32Array(pointCount);
+
+  const {iterable, objectInfo} = createIterable(data);
+  for (const object of iterable) {
+    objectInfo.index++;
+    const position = getPosition(object, objectInfo);
+    const {index} = objectInfo;
+    positions[index * 3] = position[0];
+    positions[index * 3 + 1] = position[1];
+    positions[index * 3 + 2] = position[2] || 0;
+    weights[index] = getWeight(object, objectInfo);
+  }
+
+  return {
+    positions,
+    weights
+  };
+}
+
+// true if currentBounds contains targetBounds, false otherwise
+export function boundsContain(currentBounds, targetBounds) {
+  const [currentMin, currentMax] = currentBounds;
+  const [targetMin, targetMax] = targetBounds;
+
+  if (
+    targetMin[0] >= currentMin[0] &&
+    targetMax[0] <= currentMax[0] &&
+    (targetMin[1] >= currentMin[1] && targetMax[1] <= currentMax[1])
+  ) {
+    return true;
+  }
+  return false;
+}
+
+// For given rectangle bounds generates two triangles vertices that coverit completely
+export function getTriangleVertices(opts = {}) {
+  const {xMin = 0, yMin = 0, xMax = 1, yMax = 1, addZ = false} = opts;
+
+  if (addZ) {
+    return new Float32Array([
```

Comment:
This is not necessary - you can provide a `size:2` buffer to a vec3 attribute.

Reply:
Keeping it for now, to enable `elevation`, there was one user requesting it.